scripts/publish_pr.py

```
@@ -138,6 +149,20 @@ def stage_and_commit(files: Optional[list[str]], message: str, branch: str) -> b
         return False
 
 
+def branch_is_ahead_of_main(branch: str) -> bool:
+    """Check if branch has commits ahead of main."""
+    try:
+        result = subprocess.run(
```

Ruff (0.15.17)

[error] `subprocess` call: check for execution of untrusted input

(S603)